### PR TITLE
Quantinuum rebrand

### PR DESCRIPTION
**Pull request by cgranade**
This draft PR begins changes needed to update Quantinuum targets to reflect their recent rebranding.

### CODE REVIEW

File: src/Tests/AzureClientTests.cs [hunk 1]
```diff
@@ -150,7 +156,7 @@ public void TestManualTargets()
             // set up the mock workspace
             var azureWorkspace = azureClient.ActiveWorkspace as MockAzureWorkspace;
             Assert.IsNotNull(azureWorkspace);
-            azureWorkspace?.AddProviders("ionq", "honeywell", "unrecognized");
+            azureWorkspace?.AddProviders("ionq", "quantinuum", "unrecognized");
```
Comment (vxfield) [inline src/Tests/AzureClientTests.cs:159]: should we keep `honeywell` scenario here to make sure it continues to work?

File: src/Tests/IQsharpEngineTests.cs [hunk 2]
```diff
@@ -931,7 +931,7 @@ public async Task CompileAndSubmitWithClassicalControl()
                 .Input(source)
                     .ExecutesSuccessfully()
                 .WithMockAzure()
-                .Input("%azure.target honeywell.mock")
+                .Input("%azure.target quantinuum.mock")
```
Comment (vxfield) [inline src/Tests/IQsharpEngineTests.cs:934]: should we keep `honeywell` scenario here to make sure it continues to work?

File: src/Tests/TestExtensions.cs [hunk 3]
```diff
@@ -41,7 +41,7 @@ await engine
             var client = await (await engine).Engine.GetEngineService<IAzureClient>();
             if (client is AzureClient azureClient && azureClient.ActiveWorkspace is MockAzureWorkspace workspace)
             {
-                workspace.AddProviders("ionq", "honeywell");
+                workspace.AddProviders("ionq", "quantinuum");
```
Comment (vxfield) [inline src/Tests/TestExtensions.cs:44]: should we keep `honeywell` scenario here to make sure it continues to work?

**Review by vxfield — COMMENTED** [review]
Looking good so far.

Is there any place in IQ# in which input/output-format values are used?
To recap, the Quantinuum input/output-format values will be the same as Honeywell.

File: src/Tests/AzureClientTests.cs (hunk 1 above)
Comment (cgranade) [reply src/Tests/AzureClientTests.cs:159]: I believe that's covered in other tests, while this one makes sure that IonQ works when Quantinuum neé Honeywell is also added. That said, happy to add both providers here to be sure to cover that case when using IonQ as well.

File: src/Tests/IQsharpEngineTests.cs (hunk 2 above)
Comment (cgranade) [reply src/Tests/IQsharpEngineTests.cs:936]: Fixed.

File: src/Tests/TestExtensions.cs (hunk 3 above)
Comment (cgranade) [reply src/Tests/TestExtensions.cs]: Fixed.

**Comment by cgranade** [comment]
Thanks for all your help, @cesarzc, and for your review @vxfield! The PR now works locally, so moving this out of draft and marking as ready to review.

image

**Review by cesarzc — APPROVED** [review]
LGTM!

**Pull request merged**